compiler/tools/suggest.nim

```diff
@@ -466,6 +468,48 @@ proc isTracked*(current, trackPos: TLineInfo, tokenLen: int): bool =
     if col >= current.col and col <= current.col+tokenLen-1:
       return true
 
+proc findTrackedNode(n: PNode; trackPos: TLineInfo): PSym =
+  if n.kind == nkSym:
+    if isTracked(n.info, trackPos, n.sym.name.s.len): return n.sym
+  else:
+    for i in 0 ..< safeLen(n):
+      let res = findTrackedNode(n[i], trackPos)
+      if res != nil: return res
+
+proc findTrackedSym*(g: ModuleGraph;): PSym =
+  let m = g.getModule(g.config.m.trackPos.fileIndex)
+  if m != nil and m.ast != nil:
+    result = findTrackedNode(m.ast, g.config.m.trackPos)
+
+proc executeCmd*(cmd: IdeCmd, file, dirtyfile: AbsoluteFile, line, col: int;
+             graph: ModuleGraph) =
+  let conf = graph.config
+  conf.ideCmd = cmd
+  var isKnownFile = true
+  let dirtyIdx = fileInfoIdx(conf, file, isKnownFile)
+
+  if not dirtyfile.isEmpty: msgs.setDirtyFile(conf, dirtyIdx, dirtyfile)
+  else: msgs.setDirtyFile(conf, dirtyIdx, AbsoluteFile"")
+
+  conf.m.trackPos = newLineInfo(dirtyIdx, line, col)
+  conf.m.trackPosAttached = false
+  conf.errorCounter = 0
+  if not isKnownFile:
+    graph.compileProject(dirtyIdx)
+  if conf.ideCmd in {ideUse, ideDus} and
```

Comment:
hmm, that's a bit weird because `ideDus` is definition + usage, IIRC, in which case definition (`ideDef`?) should be included... or `ideDus` shouldn't be included here. Probably not worth changing right now, mostly and observation

Reply:
yep, this piece of code is questionable, maybe other ideCmds should include too.